app/com/linkedin/drelephant/purge/AppResultPurger.java

```diff
@@ -0,0 +1,74 @@
+package com.linkedin.drelephant.purge;
+
+import com.avaje.ebean.Ebean;
+import com.avaje.ebean.QueryIterator;
+import models.AppHeuristicResult;
+import models.AppResult;
+import org.joda.time.DateTime;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class AppResultPurger {
+
+    private static final Logger logger = LoggerFactory.getLogger(AppResultPurger.class);
+
+    private static final int BATCH_SIZE = 10000;
+
+    public static int deleteOlderThan(int days) {
+
+        QueryIterator<AppResult> it = null;
+        int count = 0;
+        int loopCount = 1;
+
+        while (true) {
+
+            logger.debug("Loop " + loopCount + " - Trying to purge " + BATCH_SIZE + " app results");
+
+            Ebean.beginTransaction();
+
+            try {
+
+                it = Ebean.find(AppResult.class)
+                    .select("id")
+                    .fetch("yarnAppHeuristicResults",
+                        "id") //Fetch heuristic results ids because we will need to delete them right away
+                    .where()
+                    .lt("finishTime", DateTime.now().minusDays(days).getMillis())
```

Comment:
Assign a var before the while loop equal to DateTime.now().minusDays(days).getMillis() and reuse it for all loop.
It will aviod in extreme case to have this thread never ending.